respx/models.py

```diff
@@ -65,14 +64,42 @@
 
 DefaultType = TypeVar("DefaultType", bound=Any)
 
-Regex = type(re.compile(""))
 Kwargs = Dict[str, Any]
-URLPatternTypes = Union[str, Pattern[str], URL]
+URLPatternTypes = Union[str, Pattern[str], URL, httpx.URL]
 JSONTypes = Union[str, List, Dict]
 ContentDataTypes = Union[bytes, str, JSONTypes, Callable, Exception]
+QueryParamTypes = Union[bytes, str, List[Tuple[str, Any]], Dict[str, Any]]
 
-istype = lambda t, o: isinstance(o, t)
-isregex = partial(istype, Regex)
+
+def build_url(
+    url: URLPatternTypes, *, base: str = "", params: Optional[QueryParamTypes] = None
+) -> Union[httpx.URL, Pattern[str]]:
+    if not url:
+        if params is not None:
+            raise ValueError("Params cannot be used with empty url.")
+        return None
```

Comment:
Great, but what if we have a `base`? In that case `params` would be possible/allowed.